src/main/java/org/scijava/io/handle/FileHandle.java

```diff
@@ -306,11 +306,38 @@ public synchronized void close() throws IOException {
 
 	// -- Helper methods --
 
-	private RandomAccessFile raf() throws IOException {
+	/**
+	 * Access method for the internal {@link RandomAccessFile}, that succeeds
+	 * independently of the underlying file existing on disk. This allows us to
+	 * create a new files for writing.
+	 *
+	 * @return the internal {@link RandomAccessFile} or creates if if needed.
+	 * @throws IOException if the {@link RandomAccessFile} could not be created.
+	 */
+	private RandomAccessFile writeRaf() throws IOException {
 		if (raf == null) initRAF();
 		return raf;
 	}
 
+	/**
+	 * Access method for the internal {@link RandomAccessFile}, that only succeeds
+	 * if the underlying file exists on disk. This prevents accidental creation of
+	 * empty RAF file when calling read operations on a non-existent file.
+	 *
+	 * @return the internal {@link RandomAccessFile} or creates if if needed
+	 * @throws IOException if the {@link RandomAccessFile} could not be created,
+	 *           or the backing file does not exists.
+	 */
+	private RandomAccessFile readRaf() throws IOException {
+		if (raf == null) {
+			if (!exists()) {
```

Comment:
I would prefer this `exists()` check live in the `initRAF()` method. It can be done by making the method `initRAF(final boolean create)` and then writing `if (!create && !exists()) throw ...`. Then, here in `readRAF()`, simply put `if (raf == null) initRAF(false);`—and correspondingly, `initRAF(true)` in the `writeRAF()` method. Then the two methods will be symmetrical, and IMHO easier to understand.

Note also the case of `RAF`—it is an acronym, so should be all capitals, not camel case `Raf`. Maybe we want to call them `reader()` and `writer()` instead of `readRaf()` and `writeRaf()`? I feel like `readRaf()` and `writeRaf()` are somehow too long/unwieldy.

Reply:
Great suggestion, this makes the code much cleaner :)